### ISSUE TITLE
Research player integration and create a plan on how to carry out player integration.

### ISSUE THREAD

**Issue by victoriavillalba** [uchicago-cs/chiventure#589]
This issue has been passed from different groups and it is now out responsibility. I found a wiki that was made that fully explains what the goal of player integration should look like:
https://github.com/uchicago-cs/chiventure/wiki/%22Player%22-Implementation:-Current-Support-&-Integration-with-WDL
The first thing we should do as a group is discuss what the best plan fr this is and then create another wiki on how we should carry out player integration. For this sprint, the goal would be to do this and hopefully develop some code.

**Comment by victoriavillalba** [comment]
Possible tasks to work on:
1. only allowing specific actions to be carried out by specific classes.
2. making actions have an effect on status (e.g. eat apple -> +15 health)
3. knowing when to end the game (0 health)
4. show player stats when called for.
5. skill trees enum struct? also coordinating skills with actions

**Comment by svedwards23** [comment]
Things to keep in mind for tasks above:
- future implementation of custom actions
- interactions between actions and player attributes/skills
- future implementation of checkpointing (the result of 0 health may vary)

**Comment by ericyc16** [comment]
After meeting today to discuss ideas for player integration, we have come up with a tentative design that is outlined [here](https://github.com/uchicago-cs/chiventure/wiki/Action-Management-Player-Integration-Plan). 

Essentially, we are thinking of adding structs to define a player's class, stats, and statuses. The player struct will be updated to include a class field, race field (for now, just a string variable), stats hashtable field, and statuses list field. Skill trees will be included in the class struct.

**Comment by ericyc16** [comment]
Met with @michaelleestuy (from RPG skill trees) and @richardxd (from RPG player stats) to discuss integrating their RPG features into the player struct.

**Skill trees**: The only addition necessary to the player struct is the addition of an `allskills_t` struct. This is a container for multiple skills. On a related note, a global skill tree containing all possible skills will likely need to be added to the game struct.

Currently, the skeleton code for skill trees can be found in the `skilltrees/skeleton_code` branch, with struct definitions in `chiventure/include/skilltrees/skilltrees.h`.

**Player stats**: Two structs need to be added to the player struct. The `stats_t` struct (added as a hashtable of stats) represents a player's stats and stores (among other things) the name of a stat, its current value, and its maximum value. The `effects_t` struct (also added as a hashtable) represents effects that change a player's stats and stores the name of the effect, whether or not its active, and a list of affected stats.

Currently, the code for player stats can be found in the `rpg-stats` branch, with struct definitions in `chiventure/include/game-state/stats.h`. Note that, on this branch, the RPG stats team has already gone ahead and added the two structs to a player, so we wouldn't have to actually do that work. However, it's good for us to still be aware of this.

**Comment by ericyc16** [comment]
Even though I have not yet been able to meet with the RPG player class team, I found [this wiki](https://github.com/uchicago-cs/chiventure/wiki/Player-Class:-Design-Plan) detailing their design. I have also found that they have their class struct merged with the `dev` branch. Their definitions can be found in `include/playerclass/class_structs.h`. It looks like the only thing necessary from their end for us to add to the player struct is a `class_t` field.

I have updated [the wiki](https://github.com/uchicago-cs/chiventure/wiki/Action-Management-Player-Integration-Plan) based on this information and based on yesterday's meeting with the RPG skill trees and RPG player stats teams. It seems like the only thing not accounted for by the RPG teams is some way to define player race. For now, I have added it to the `player_t` struct simply as a `char*` field.
